.github/workflows/on-every-commit.yml

```diff
@@ -0,0 +1,27 @@
+name: On Every Commit
+
+on: push
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  unit-test:
+    strategy:
+      fail-fast: false
+      matrix:
+        # macos-14 is M1 (i.e., ARM) and macos-13 is Intel (i.e., x86)
+        os: [ubuntu-latest, macos-14, macos-13]
+    runs-on: ${{ matrix.os }}
+    steps:
+    - name: Checkout Repo
+      uses: actions/checkout@v3
+
+    - name: Setup Environment
+      uses: ./.github/actions/setup-env
+      with:
+        go: 'true'
```

Comment:
Once you add the go version file flag this is how you would pass it:
```yaml
go-version-file: './go.mod'
```

Reply:
done